plutus-pab/src/Plutus/PAB/Run.hs

```diff
@@ -0,0 +1,88 @@
+{-# LANGUAGE ApplicativeDo         #-}
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE DerivingVia           #-}
+{-# LANGUAGE FlexibleContexts      #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE NamedFieldPuns        #-}
+{-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE RankNTypes            #-}
+{-# LANGUAGE ScopedTypeVariables   #-}
+
+module Plutus.PAB.Run
+( runWith
+) where
+
+import qualified Cardano.BM.Backend.EKGView          as EKGView
+import qualified Cardano.BM.Configuration.Model      as CM
+import           Cardano.BM.Data.Trace               (Trace)
+import           Cardano.BM.Plugin                   (loadPlugin)
+import           Cardano.BM.Setup                    (setupTrace_)
+import           Control.Concurrent.Availability     (newToken)
+import           Control.Monad                       (when)
+import           Control.Monad.IO.Class              (liftIO)
+import           Control.Monad.Logger                (logErrorN, runStdoutLoggingT)
+import           Data.Aeson                          (FromJSON, ToJSON)
+import           Data.Foldable                       (for_)
+import           Data.Text.Extras                    (tshow)
+import           Data.Typeable                       (Typeable)
+import           Data.Yaml                           (decodeFileThrow)
+import           Plutus.PAB.Effects.Contract.Builtin (Builtin, BuiltinHandler, HasDefinitions)
+import           Plutus.PAB.Monitoring.Config        (defaultConfig, loadConfig)
+import           Plutus.PAB.Monitoring.PABLogMsg     (AppMsg (..))
+import           Plutus.PAB.Monitoring.Util          (PrettyObject (..), convertLog)
+import           Plutus.PAB.Run.Cli
+import           Plutus.PAB.Run.CommandParser
+import           Plutus.PAB.Run.PSGenerator          (HasPSTypes)
+import           Plutus.PAB.Types                    (PABError (MissingConfigFileOption))
+import           Prettyprinter                       (Pretty)
+import qualified Servant
+import           System.Exit                         (ExitCode (ExitFailure), exitSuccess, exitWith)
+
+runWith :: forall a.
+    ( Show a
+    , Ord a
+    , FromJSON a
+    , ToJSON a
+    , Pretty a
+    , Servant.MimeUnrender Servant.JSON a
+    , Typeable a
+    , HasDefinitions a
+    , HasPSTypes a
+    )
+    => BuiltinHandler a
+    -> IO ()
+runWith userContractHandler = do
```

Comment:
Needs haddocks

Something like "PAB entry point for a contract type `a`"

Reply:
Added some comments. Adequate enough?